core/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -355,7 +355,8 @@ class Router(formatOps: FormatOps) {
           case _ => NoSplit
         }
         Seq(
-          Split(modification, 0)
+          Split(modification, 0),
```

Comment:
The failing tests with "Search state exploded" are because there are two possible splits here. Each split creates a new branch that is explored by the search algorithm, similar to how [A*](https://en.wikipedia.org/wiki/A*_search_algorithm) works. 

You can remove the line below, and update instead `modification` , which is defined above.